examples/plotting/plot_4_partial_dependencies.py

```diff
@@ -0,0 +1,207 @@
+value of the grid. For instance, for a 1-D grid of `g` points and a group of `n` points, we compute
+`g * n` predictions.
+
+For a search space of `d` hyperparameters, the partial dependency plot is organized as a matrix of 
+`(d, d)` subplots. The subplots on the diagonal show the partial dependency of each hyperparameters
+separately, while the subplots below the diagonal show the partial dependency between two
+hyperparameters. Let's look at a simple example to make it more concrete.
+
+.. autofunction:: orion.plotting.base.partial_dependencies
+    :noindex:
+
+The partial dependencies plot can be executed directly from the ``experiment`` with
+``plot.partial_dependencies()`` as shown in the example below.
+
+"""
+from orion.client import get_experiment
+
+# Specify the database where the experiments are stored. We use a local PickleDB here.
+storage = dict(type="legacy", database=dict(type="pickleddb", host="../db.pkl"))
+
+# Load the data for the specified experiment
+experiment = get_experiment("2-dim-exp", storage=storage)
+fig = experiment.plot.partial_dependencies()
+fig
+
+#%%
+# For the plots on the diagonal, the y-axis is the objective and the x-axis is the value of the
+# corresponding hyperparameter. For the contour plots below the diagonal, the y-axis and x-axis are
+# the values of the corresponding hyperparameters labelled on the left and at the bottom. The
+# objective is represented as a color gradient in the contour plots. The light blue area in
+# the plots on the diagonal represents the standard deviation of the predicted objective when
+# varying the other hyperparameters over the search space.
+# The black dots represents the
+# trials in the current history of the experiment. If you hover your cursor over one dot, you will
+# see the configuration of the corresponding trial following this format:
+#
+# ::
+#
+#    ID: <trial id>
+#    value: <objective>
+#    time: <completed time>
+#    parameters
+#      <name>: <value>
+#
+# Even for a simple 2-d search space, the partial dependency is very useful. We see very cleary in this example
+# the optimal regions for both hyperparameters and we can see as well that the optimal region for
+# learning rates is larger when the dropout is low, and narrower when dropout approaches 0.5.
+#
+# .. TODO::
+#
+#     Make one toy example where two HPs are dependent.
+#
+
+#%%
+# Options
+# -------
+#
+# Params
+# ~~~~~~
+#
+# The simple example involved only 2 hyperparameters, but typical search spaces can be much larger.
+# The partial dependency plot becomes hard to read with more than 3-5 hyperparameters dependency on
+# the size of your screen. With a fix width like in this documentation, 5 hyperparameters are
+# impossible to read as you can see below.
+# (Data coming from tutorial :ref:`sphx_glr_auto_tutorials_code_2_hyperband_checkpoint.py`)
+
+experiment = get_experiment("hyperband-cifar10", storage=storage)
+experiment.plot.partial_dependencies()
```

Comment:
The plot generated here has several data points represented by dots, which makes the plots look quite busy and obscures the underlying contours. Is it possible to optionally remove the dots representing the data points?

Reply:
I could change the size of the dots, or make them optional.